examples/python/braket.py

```diff
@@ -0,0 +1,124 @@
+# ## AWS via pytket
+
+# This notebook describes how to how to run a simple circuit using the `pytket-braket` backend from scratch.
+
+# ### Prerequisites
+
+# #### Python packages:
+
+# - `pytket`'s extension for Amazon Braket devices [pytket-braket](https://pypi.org/project/pytket-braket/)
+#     - `pip install pytket-braket`
+# - Amazon's braket package [amazon-braket-sdk](https://pypi.org/project/amazon-braket-sdk/)
+#     - `pip install amazon-braket-sdk`
+# - `boto3`, which is Amazon's package for accessing AWS (so called because Boto dolphins live in the Amazon 🙄) [boto3](https://pypi.org/project/boto3/)
+#     - `pip install boto3`
+
+# ### What are we doing?
+
+# There is a hierarchy of things we need to access here, as illustrated below. Amazon Web Services (AWS) contains many things, one of which is Amazon Braket. That in turn contains many quantum devices we can run things on.
+
+# Essentially we nest interfaces from the three packages above to access each layer of services:
+# $$ \textrm{amazon web services} \rightarrow \textrm{amazon braket} \rightarrow \textrm{quantum device} $$
+
+# <div>
+# <img src="python/aws_img.png" width="500"/>
+# </div>
+
+# ### Credentials
+
+# You need the following credentials to access devices with AWS:
+# - AWS account ID (12 digits) or account alias
+# - User name
+# - Password
+
+# And for accessing services in python, a public and private key:
+# - AWS access key ID
+# - AWS secret access key
+
+
+# ### What is S3?
+
+# Amazon has a Dropbox-like service called Amazon S3 -- Simple Storage Service. This is a service under the AWS umbrella where results from experiments can be saved. These are saved in top-level folders called 'S3 buckets'.
+
+
+# ### Setting up AWS
+
+# #### Checking the region for the device you want
+# - Find Amazon Braket and go to 'Devices'
+# - Check the `region` for the device you want to access (e.g. for OQC's Lucy device, this is `eu-west-2`)
+
+# #### Create an S3 bucket
+# - Under AWS, go to Amazon S3 and create a new bucket and choose a `name` and the `region` from above. __NOTE:__ the `name` __must__ start with the prefix `amazon-braket`, e.g. `amazon-braket-my-test-experiment`!
+# - Once you've created the bucket, select it and create a folder inside. In python, the name of this folder becomes known as the `bucket_key`.
+
+# #### Create a credentials file
```

Comment:
There is another way to do this, as described here: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/quickstart.html

Following the instructions there actually does create a couple of files in `~/.aws` containing credentials and config. I think it is better to stick to the AWS recommendations; if we advise users to put their credentials in a new file like this then we also need to warn them about the security implications ... I would rather leave that responsibility to AWS!

After setting up the credentials as described there, it is enough to set a default `region`, `s3_bucket` and `s3_folder` in pytket config. For example, my `~/.config/pytket/config.json` contains

```
    "braket": {
      "region": "us-west-1",
      "s3_bucket": "amazon-braket-test",
      "s3_folder": "test-folder"
    },
```

Reply:
Thanks for the comments, yes, that makes sense. I'm having trouble using the `config` file it seems -- should it be enough to create the file as above (with `"eu-west-2"` for the example I'm using) and then just call ```
aws_oqc_backend = BraketBackend(
    local=False,
    device="Lucy",
    device_type="qpu",
    provider="oqc",
    aws_session=my_aws_session,
)
```
I'm getting `ValueError: QPU 'arn:aws:braket:::device/qpu/oqc/Lucy' not found`, although when I pass `region`, `s3_bucket` and `s3_folder` manually it works fine -- suggests I might not be using the config file right.

Reply:
The full file would be
```json
{
  "extensions": {
    "braket": {
      "region": "us-west-1",
      "s3_bucket": "amazon-braket-test",
      "s3_folder": "test-folder"
    }
}
```

Reply:
Sorry, yes, that's what I've got:
`{
  "extensions": {
    "braket": {
      "s3_bucket": "amazon-braket-ox-test-eu-west-2",
      "s3_folder": "test_folder",
      "region": "eu-west-2"
    }
  }
}`

Reply:
Ah yes that's the one, thanks -- all good